### PR TITLE
luooMusic:  Add support for essay, musician, search

### PR DESCRIPTION
**Pull request by xyauhideto**

### CODE REVIEW

File: connectors/luoo.js [hunk 1]
```diff
@@ -1,9 +1,109 @@
 'use strict';
+(() => {
```
Comment (alexesprit) [inline connectors/luoo.js]: Is there a reason to pull all code in this self-invoking function?

File: connectors/luoo.js [hunk 2]
```diff
@@ -1,9 +1,109 @@
...
+	let setupConnector = () => {
```
Comment (alexesprit) [inline connectors/luoo.js]: Could you use function declaration instead of function expression?

Comment (xyauhideto) [reply connectors/luoo.js]: Okay

File: connectors/luoo.js [hunk 3]
```diff
@@ -1,9 +1,109 @@
...
+			let text = Connector.getCurrentItem().find('.content').text();
+			let [album, artist] = text.trim().split(', ').map((text) => text.replace(/^\S+\s/g, '')) || Array(2).fill(null);
+			return { album, artist };
+		};
+
+		Connector.getAlbum = () => Connector.getAlbumArtist().album;
+
+		Connector.getArtist = () => Connector.getAlbumArtist().artist;
+
+		Connector.getUniqueID = () => Connector.getCurrentItem().data('id') || null;
```
Comment (alexesprit) [inline connectors/luoo.js]: `|| null` can be omitted.

File: tests/connectors/luoo.js [hunk 4]
```diff
@@ -1,7 +1,21 @@
...
+		url: 'http://www.luoo.net/essay/915'
+	});
+	// musician
+	connectorSpec.shouldBehaveLikeMusicSite(driver, {
+		url: 'http://www.luoo.net/musician/'
+	});
+	// search
+	connectorSpec.shouldBehaveLikeMusicSite(driver, {
+		url: 'http://www.luoo.net/search/?q=love&t=single',
+		playButtonSelector: '.icon-search-play:eq(-1)'
```
Comment (alexesprit) [inline tests/connectors/luoo.js]: WebDriver throws an error when clicks on an element with such selector. Do you have the same issue?

I tried to use `.icon-search-play` selector and it works fine for me, but there's another thing that first three songs from search results are unplayable for me. Clicking on play icon does nothing. I updated the test like this way and it passed for me:
```js
	// search
	connectorSpec.shouldBehaveLikeMusicSite(driver, {
		url: 'http://www.luoo.net/search/?q=Spinning&t=single',
		playButtonSelector: '.icon-search-play'
	});
```

**Pull request merged**